sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LocalRelation.scala

```diff
@@ -93,4 +94,12 @@ case class LocalRelation(
   }
 
   override def maxRows: Option[Long] = Some(data.length.toLong)
+
+  override val nodePatterns: Seq[TreePattern] = {
+    if (this.data.isEmpty) {
```

Comment:
shall we just make it `LOCAL_RELATION`?
We can prune `LOCAL_RELATION` in `ConvertToLocalRelation`.
We got general patterns like `BINARY_COMPARISON` and `BINARY_ARITHMETIC`. This one is very specific and it looks odd.

Reply:
Done. 
Thanks, Gengliang! Make sense!